### ISSUE TITLE
kernel injection not working (vmt version)

### ISSUE THREAD

**Issue by beranos** [mactec0/Kernelmode-manual-mapping-through-IAT#7]
hi, i tried the kernel injector this version (https://github.com/mactec0/Kernelmode-manual-mapping-through-IAT/tree/swapchain_vmt_example)

i'm testing on RainbowSix when i open the injector the console says everything is fine even tho the dll is not being injected (using testdll)

EDIT: Oh, nevermind. i figured out GameOverlayRenderer64.dll is from steam overlay but i'm using the uplay version. my bad
anyways now it's saying "Cannot find vmt table" so i'm guessing the offsets changed?

**Comment by reahly** [comment]
> hi, i tried the kernel injector this version (https://github.com/mactec0/Kernelmode-manual-mapping-through-IAT/tree/swapchain_vmt_example)
> 
> i'm testing on RainbowSix when i open the injector the console says everything is fine even tho the dll is not being injected (using testdll)
> 
> EDIT: Oh, nevermind. i figured out GameOverlayRenderer64.dll is from steam overlay but i'm using the uplay version. my bad
> anyways now it's saying "Cannot find vmt table" so i'm guessing the offsets changed?

yes it changed, https://i.imgur.com/fKLxlcc.png
1800AEECB - imagebase = 0x18D4C8

**Comment by Bix3** [comment]
for me the game crashes with the offset @dollarssign  posted (0x18D4C8) tough i can confirm its the right one when i scan the sig i get the same. it crashes on: https://github.com/mactec0/Kernelmode-manual-mapping-through-IAT/blob/dd1d848a380c9496cbdbafc0225790bd09ba0d52/mmap/mmap.cpp#L230

**Comment by reahly** [comment]
> for me the game crashes with the offset @dollarssign posted (0x18D4C8) tough i can confirm its the right one when i scan the sig i get the same. it crashes on:
> 
> https://github.com/mactec0/Kernelmode-manual-mapping-through-IAT/blob/dd1d848a380c9496cbdbafc0225790bd09ba0d52/mmap/mmap.cpp#L230

are you sure? it works fine for me with the test dll attached to the project

**Comment by Bix3** [comment]
yes it works now, i compiled the dll from source and didnt saw that there were project files, thank you!

**Comment by reahly** [comment]
changed again, new offset: 0x18E528

**Comment by Grabowski95** [comment]
hi there,
thanks for the splendit work but i cant manage it to work (testing on warhammer vermintide 2 mod realm). I grabbed new offset 0x01B7528 but seems its not correct. It says injected but no messagebox shows up.
Also, where did you get these 2 pointers 0xE0 and 0x00?

**Comment by ThaFurther** [comment]
>Also, where did you get these 2 pointers 0xE0 and 0x00?

Did you get an answer on this?

**Comment by GerritdinaH** [comment]
Not working for me as well, i found the right Offset for the swapchain, but its sadly not working :/

**Comment by GerritdinaH** [comment]
well got it working by replacing 0xE0 with 0xF8, but im pretty sure thats not the correct way since it crashes as soon as i press enter, so it crashes when setting the protection flag of the VMT-Function to PAGE_READONLY

**Comment by GerritdinaH** [comment]
anyone found a Swapchain pointer for DiscordHook64?
